### PR TITLE
feat(api-client): Create controller for Secret module

### LINKED ISSUE #352: API-CLIENT: Create controller for Secret module
## Description 

Refer to #347 

API Controller: https://github.com/keyshade-xyz/keyshade/blob/develop/apps/api/src/secret/controller/secret.controller.ts

### PR DESCRIPTION
**Pull request by vr-varad**
## Description

Api client Controller for Secret Module

Fixes #352 

## Dependencies

_Mention any dependencies/packages used_

## Future Improvements

_Mention any improvements to be done in future related to any file/feature_

## Mentions

_Mention and tag the people_

## Screenshots of relevant screens

_Add screenshots of relevant screens_

## Developer's checklist

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-check on my work

**If changes are made in the code:**

- [x] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [x] My changes in code generate no new warnings
- [ ] My changes are breaking another fix/feature of the project
- [ ] I have added test cases to show that my feature works
- [ ] I have added relevant screenshots in my PR
- [ ] There are no UI/UX issues

### Documentation Update

- [ ] This PR requires an update to the documentation at [docs.keyshade.xyz](https://docs.keyshade.xyz)
- [ ] I have made the necessary updates to the documentation, or no documentation changes are required.

### CODE REVIEW

**Comment by vr-varad** [comment]
@rajdip-b Can u give it a look whenever you are free.

File: packages/api-client/src/types/secret.types.d.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+export interface GetAllSecretsOfEnvironmentRequest {
+  projectId: string
+  environmentId: string
+  page?: number
+  limit?: number
+  sort?: string
+  order?: string
+  search?: string
+}
+export interface GetAllSecretsOfEnvironmentResponse {
```
Comment (rajdip-b) [inline packages/api-client/src/types/secret.types.d.ts:106]: Please check the return type from postman or make a query to this endpoint yourself.

File: packages/api-client/src/types/secret.types.d.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  entries?: [
+    {
+      value: string
+      environmentId: string
+    }
+  ]
+}
+
+export interface UpdateSecretResponse {
+  secret: any
```
Comment (rajdip-b) [inline packages/api-client/src/types/secret.types.d.ts:48]: Refrain from using `any`

File: packages/api-client/src/types/secret.types.d.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
+export interface CreateSecretRequest {
+  projectId: string
+  name: string
+  note?: string
+  rotateAfter?: string
```
Comment (rajdip-b) [inline packages/api-client/src/types/secret.types.d.ts:5]: copy the exact type of rotateAfter from the dto in api

**Review by rajdip-b — CHANGES_REQUESTED** [review]

File: packages/api-client/src/types/secret.types.d.ts [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+export interface GetAllSecretsOfProjectRequest {
+  projectId: string
+  page?: number
+  limit?: number
+  sort?: string
+  order?: string
+  search?: string
+}
+export interface GetAllSecretsOfProjectResponse {
```
Comment (rajdip-b) [inline packages/api-client/src/types/secret.types.d.ts:85]: The data should be returned in `items`. Modify the function and the type accordingly.

File: packages/api-client/src/types/secret.types.d.ts [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    }
+    value: string
+    version: number
+  }
+}
+
+export interface GetAllSecretsOfEnvironmentRequest {
+  projectId: string
+  environmentId: string
+  page?: number
```
Comment (rajdip-b) [inline packages/api-client/src/types/secret.types.d.ts:113]: This endpoint wont have pagination

**Review by rajdip-b — CHANGES_REQUESTED** [review]

**Comment by rajdip-b** [comment]
@vr-varad can you start with this pr? please make the changes as we have done in environments controller.

**Comment by vr-varad** [comment]
@rajdip-b Yeah Sure

**Comment by vr-varad** [comment]
@rajdip-b Can u review it whenever u are free

**Review by rajdip-b — APPROVED** [review]
LGTM

**Pull request merged**